sys/parsoid.js

```diff
@@ -159,11 +159,42 @@ function _dependenciesUpdate(hyper, req) {
     });
 }
 
+function compileReRenderBlacklist(blacklist) {
+    const result = {};
+
+    /**
+     * From a list of regexes and strings, constructs a regex that
+     * matches any of list items
+     */
+    const constructRegex = (variants) => {
+        let regex = (variants || []).map((regexString) => {
+            regexString = regexString.trim();
+            if (/^\/.+\/$/.test(regexString)) {
+                return `(?:${regexString.substring(1, regexString.length - 1)})`;
+            }
+            // Compare strings, instead
+            return `(?:^${decodeURIComponent(regexString)
+                .replace(/[-[\]/{}()*+?.\\^$|]/g, "\\$&")})`;
```

Comment:
We want to match the whole string, not just its beginning, so you are missing a `$` at the end